src/main/kotlin/Application.kt

```diff
@@ -14,6 +16,33 @@ fun main() {
     val gameSize = GameSize(mapWidth, mapHeight)
     val minePositions = MinePositionGenerator(gameSize, mineCount).generateMinePositions()
     val gameMap = GameMap.from(gameSize, minePositions)
-    gameMap.setNormalBlocks()
-    showMap(gameSize, gameMap)
+    Output.startGame()
+    Output.showMap(gameMap)
+
+    val openResult = pickOpen(gameMap)
+    Output.printResult(openResult.message)
+}
+
+fun pickOpen(gameMap: GameMap): OpenResult {
+    var openResult = OpenResult.CONTINUE
+    while (openResult == OpenResult.CONTINUE) {
+        val openPosition = openBlock()
+        val alreadyOpened = gameMap.checkAlreadyOpened(openPosition)
+        if (alreadyOpened) {
+            Output.alreadyOpenPosition()
+            continue
+        }
+        gameMap.open(openPosition)
+
+        openResult = gameMap.checkOpenedMineBlock()
+        if (openResult == OpenResult.LOSE) {
+            break
+        }
+        openResult = gameMap.isWin()
+        if (openResult == OpenResult.WIN) {
+            break
+        }
```

Comment:
조건문 내부에서 동일하게 break를 하는데 아래처럼 조건문을 하나로 합치는건 어떨까요?

```suggestion
        openResult = gameMap.??()
        if (openResult ==LOSE || openResult == WIN) {
            break
        }```

Reply:
조언 감사합니다!